wps-cloudformation-template.yaml

```diff
@@ -57,7 +57,44 @@ Parameters:
   geoserverWpsEndpoint:
     Type: String
     Default: 'http://geoserver-wps.aodn.org.au/geoserver/wps'
+  environment:
+    Type: String
+    Default: dev
+    AllowedValues:
+      - production
+      - systest
+      - sandbox
+      - dev
+  createProdSystestBucket:
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+Mappings:
+  EnvironmentMap:
+    production:
+      BucketName: "aws-wps-production"
+    systest:
+      BucketName: "aws-wps-systest"
+Conditions:
+  CreateProdSystestBucket: !And
+    - !Equals [!Ref createProdSystestBucket, true]
+    - !Not [!Condition CreateSandboxDevBucket]
+  CreateSandboxDevBucket: !Not [!Or [!Equals [ !Ref environment, production ], !Equals [ !Ref environment, systest ]]]
 Resources:
+  S3BucketProdSystest:
```

Comment:
One more thought... this seems to create a bucket, but I can't see where anything  *uses* the bucket? There are other parameters relating to bucket names, so shouldn't they be replaced with this bucket creation in CF now?

Also, given there are two possible resource names for the bucket, won't everything else need logic to determine which resource name to reference, based on the environment parameter?

For example, anything referencing the bucket will need to work out whether to reference "S3BucketProdSystest" or "S3BucketSandboxDev", because only one of them will exist...

Reply:
You are right. We will need to figure out a way to reference the bucket. For this task we were suppose to add s3 to cloudformation.

Reply:
> For this task we were suppose to add s3 to cloudformation.

I think it was implied that the bucket would be created AND used by the stack... ain't nobody got time for a randomly dangling bucket...